spec/acceptance/httpclient/httpclient_spec.rb

```diff
@@ -113,13 +113,8 @@ def filter_response(request, response)
         nil # to let the request be made for real
       end
 
-      # To make two requests that have the same request signature, the headers must match.
-      # Since the webmock server has a Set-Cookie header, the 2nd request will automatically
-      # include a Cookie header (due to how httpclient works), so we have to set the header
-      # manually on the first request but not on the 2nd request.
-      http_request(:get, webmock_server_url, :client => client,
-                         :headers => { "Cookie" => "bar=; foo=" })
-      http_request(:get, webmock_server_url, :client => client)
+      http_request(:get, webmock_server_url, :client => client, :headers => { "Cookie" => "bar=; foo=" })
```

Comment:
This looks like it is actually an order dependent test issue. As the spec is passing if run in isolation, or even if you just run this whole file, but fails in the whole suite.

Where as your change passes in the whole suite, but fails in isolation.
I'm trying to track down the issue now, but if please feel free to jump in, you have any ideas.